src/sql/workbench/common/editorReplacerContribution.ts

```diff
@@ -63,22 +70,49 @@ export class EditorReplacementContribution implements IWorkbenchContribution {
 				editor.setMode(defaultInputCreator[0]);
 				const newInput = defaultInputCreator[1].convertInput(editor);
 				if (newInput) {
-					return { override: this.editorService.openEditor(newInput, options, group) };
+					const profile = getCurrentGlobalConnection(this.objectExplorerService, this.connectionManagementService, this.editorService);
```

Comment:
I don't think it makes sense for this code to know about connections. The input creators themselves should handle creating the inputs they need, including if it needs any connections.

Reply:
Hrm, how are you thinking this would be better handled? Have convertInput return a set of options as well or something? The editorReplacer would still need to know about making the connections in that case which seems like basically the same thing.

Reply:
Why would it need to return anything? Couldn't the convertinput method just make the connection?

Reply:
There's no editor at that point to connect

Reply:
Hm, maybe I'm misunderstanding openEditor. I'll check - but is the editor created at that point and that call is just bringing it into view?

Reply:
We never actually connect the "editors" (the actual windows themselves) we connect the inputs.